### ISSUE TITLE
xaker-eval is unreachable: no xaker-train path produces a checkpoint the CLI can load

### ISSUE THREAD

**Issue by sachncs** [sachncs/xaker#31]
## Problem

`xaker-eval` (`xaker/cli/eval.py`) loads a checkpoint with:

```python
state = torch.load(args.checkpoint, map_location=device, weights_only=True)
model.load_state_dict(state)
```

But the only training CLI (`xaker/cli/train.py`) and the in-tree trainer (`xaker/training/trainer.py`) never call `torch.save` and do not expose a checkpoint path. `Trainer.step` / `Trainer.epoch` mutate parameters in memory only. Repository-wide grep confirms `torch.save` is referenced only inside `xaker/datasets/__init__.py` (for the WikiText cache).

The CLI is therefore functionally unwired: invoking `xaker-eval --checkpoint <file>` always fails with `FileNotFoundError`, regardless of how a user has trained a model through the project's own CLIs.

## Evidence

- `xaker/cli/eval.py:26-39` — `--checkpoint` is `required=True`; the body calls `torch.load(args.checkpoint)` and `model.load_state_dict(state)`.
- `xaker/cli/eval.py:38` — hardcodes `attention_type="fused_v2"` (which is `kind_map`-aliased to `"fused"` in `xaker/model/block.py:199`).
- `xaker/cli/train.py:34-80` — `main()` builds a model and calls `Trainer.epoch(loader)`; no `torch.save` call, no `--out`/`--save` flag.
- `xaker/training/trainer.py:155-265` — `__init__`, `step`, `epoch`, `eval` only mutate params.
- `git grep -n "torch.save" xaker/` returns only `xaker/datasets/__init__.py:141` (WikiText cache).
- `xaker/training/trainer.py:130` docstring notes the trainer "does not implement… checkpointing" — but offers no mechanism to enable it.

The `tests/test_cli_full.py::test_eval_import` only checks the CLI is importable; it does not round-trip a checkpoint.

## Root Cause

The checkpoint workflow was never closed. `xaker-eval` exists but no path in the repository produces a checkpoint it can load. The CHANGELOG (0.4.0 "Paper-worthiness rubric", 0.5.0 etc.) does not mention the eval CLI being wired up.

## Impact

- Any user following README installation + quick-start cannot get from `xaker-train` outputs to `xaker-eval` inputs without writing their own save logic.
- The CLI is listed in `pyproject.toml:60-64` (`xaker-train / xaker-eval / xaker-bench / xaker-validate`) as a first-class entry point, so the broken pipeline is on the project's outer surface.
- The README Quick Start and `docs/getting-started.md` reference `xaker-eval --checkpoint artifacts/last.pt`, but `xaker-train` never writes `artifacts/`.

## Recommendation

1. Add checkpoint output to `xaker-train`:
   ```python
   parser.add_argument("--out", default=None,
                       help="Path to write the trained state_dict; required to feed xaker-eval.")
   if args.out:
       torch.save(model.state_dict(), Path(args.out))
   ```
2. Either (a) remove `xaker-eval` until it has a producer, or (b) ship a tiny `examples/save_and_load.py` showing the round-trip so the CLI is reachable.
3. Make `xaker-eval`'s hardcoded `attention_type="fused_v2"` configurable via `--kind` with the same choices as `xaker-train`.

## Acceptance Criteria

- [ ] `xaker-train --out /tmp/m.pt --dim 32 --heads 2 --epochs 1 ...` writes a loadable state-dict file.
- [ ] `xaker-eval --checkpoint /tmp/m.pt --kind standard ...` loads the file and exits 0.
- [ ] A regression test in `tests/test_cli_full.py` performs the full save+load round-trip.

## Verification

```bash
python -m xaker.cli.train --dim 32 --heads 2 --layers 1 --epochs 1 --out /tmp/m.pt --length 8 --samples 8 --batch 4 --vocab 10
python -m xaker.cli.eval --checkpoint /tmp/m.pt --length 8 --batch 2 --vocab 10
```

Both must exit 0.

## Dependencies

None.

## Risk

Low. Adds one optional `--out` flag to `xaker-train`, opens a CLI path that was previously dead.
